ocs_ci/deployment/ibmcloud.py

```diff
@@ -111,3 +121,217 @@ def check_cluster_existence(self, cluster_name_prefix):
             if cl["state"] != "deleting" and cl["name"].startswith(cluster_name_prefix)
         ]
         return bool(non_term_clusters_with_prefix)
+
+
+class IBMCloudIPI(CloudDeploymentBase):
+    """
+    A class to handle IBM Cloud IPI specific deployment
+    """
+
+    DEFAULT_STORAGECLASS = "ibmc-vpc-block-10iops-tier"
+    OCPDeployment = IPIOCPDeployment
+
+    def __init__(self):
+        self.name = self.__class__.__name__
+        self.installer = None
+        super(IBMCloudIPI, self).__init__()
+        self.credentials_requests_dir = os.path.join(self.cluster_path, "creds_reqs")
+        self.pull_secret_path = os.path.join(constants.TOP_DIR, "data", "pull-secret")
+
+    def deploy_ocp(self, log_cli_level="DEBUG"):
+        """
+        Perform IBMCloudIPI OCP deployment.
+
+        Args:
+            log_cli_level (str): log level for installer (default: DEBUG)
+        """
+        self.ocp_deployment = self.OCPDeployment()
+        self.ocp_deployment.deploy_prereq()
+
+        # IBM Cloud specific prereqs
+        ibmcloud.login()
+        self.configure_cloud_credential_operator()
+        self.export_api_key()
+        self.manually_create_iam_for_vpc()
+
+        self.ocp_deployment.deploy(log_cli_level)
+        # logging the cluster UUID so that we can ask for its telemetry data
+        cluster_id = exec_cmd(
+            "oc get clusterversion version -o jsonpath='{.spec.clusterID}'"
+        )
+        logger.info(f"clusterID (UUID): {cluster_id}")
+
+    def destroy_cluster(self, log_level="DEBUG"):
+        """
+        Destroy the OCP cluster.
+
+        Args:
+            log_level (str): log level openshift-installer (default: DEBUG)
+
+        """
+        self.export_api_key()
+        logger.info("Destroying the IBM Cloud cluster")
+        super(IBMCloudIPI, self).destroy_cluster(log_level)
+        self.delete_service_id()
+        resource_group = self.get_resource_group()
+        self.delete_volumes(resource_group)
+        time.sleep(30)  # Volumes take time to fully delete
+        self.delete_resource_group(resource_group)
+
+    def manually_create_iam_for_vpc(self):
+        """
+        Manually specify the IAM secrets for the cloud provider
+        """
```

Comment:
Empty line before the end of docstring is not consistent.

Reply:
Generally we include the empty line if it is more than a one liner and omit it if the docstring is just one line (description). There are some inconsistencies with other docstrings though so I will address those.